components/web_discovery/browser/credential_manager.cc

```diff
@@ -0,0 +1,440 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/credential_manager.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/functional/bind.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "base/logging.h"
+#include "base/task/sequenced_task_runner.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/web_discovery/browser/anonymous_credentials/rs/cxx/src/lib.rs.h"
+#include "brave/components/web_discovery/browser/pref_names.h"
+#include "brave/components/web_discovery/browser/util.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/scoped_user_pref_update.h"
+#include "crypto/sha2.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr char kJoinPath[] = "/join";
+constexpr char kJoinContentType[] = "application/json";
+
+constexpr char kJoinDateField[] = "ts";
+constexpr char kJoinMessageField[] = "joinMsg";
+constexpr char kJoinRSAPublicKeyField[] = "pk";
+constexpr char kJoinRSASignatureField[] = "sig";
+constexpr char kJoinResponseField[] = "joinResponse";
+
+constexpr char kGSKDictKey[] = "gsk";
+constexpr char kCredentialDictKey[] = "credential";
+
+constexpr net::NetworkTrafficAnnotationTag kJoinNetworkTrafficAnnotation =
+    net::DefineNetworkTrafficAnnotation("wdp_join", R"(
+    semantics {
+      sender: "Brave Web Discovery HPNv2 Join"
+      description:
+        "Retrieves anonymous credentials in order to sign Web Discovery
+         measurements sent via the HumanWeb Proxy Network."
```

Comment:
I don't believe we use the HumanWeb Proxy Network, do we?
```suggestion
         measurements."
```

Reply:
`collector.wdp.brave.com` is an HPN server